examtool/api/scramble.py

```diff
@@ -85,6 +88,20 @@ def get_elements(group):
 
 def select(substitutions):
     out = {}
+    # DEFINE
     for k, v in sorted(substitutions.items()):
         out[k] = random.choice(v)
     return out
+
+def select_no_replace(substitutions_match):
+    out = {}
+    # DEFINE MATCH
+    for item in substitutions_match:
+        k = item["directives"]
+        v = item["replacements"]
+        values = v.copy()
```

Comment:
Why do you `.copy()`? `scramble` is destructive anyway.

Reply:
I think it was because I didn't want to directly modify the other list just incase it is used elsewhere.